runtime_api.go

```diff
@@ -0,0 +1,225 @@
+package main
+
+import (
+	"go/ast"
+	"go/token"
+)
+
+// addRuntimeAPI exposes additional functions in the runtime
+// package that may be helpful when hiding information
+// during execution is required.
+func addRuntimeAPI(filename string, file *ast.File) {
+	switchPanicPrints := func(node ast.Node) bool {
+		call, ok := node.(*ast.CallExpr)
+		if !ok {
+			return true
+		}
+		id, ok := call.Fun.(*ast.Ident)
+		if !ok {
+			return true
+		}
+
+		if id.Name == "print" {
+			id.Name = "panicprint"
+			return false
+		}
+
+		return true
+	}
+
+	switch filename {
+	case "debug.go":
+		// Add HidePanics function and internal hidePanics variable
+		file.Decls = append(file.Decls, hidePanicsDecls...)
+	case "error.go":
+		// Add a function panicprint, that does nothing if panics are
+		// hidden, otherwise forwards arguments to printany to print them
+		// as normal.
+		//
+		// We will also add two statements to printany:
+		// 1. An if statement that returns early if panics are hidden, as
+		//    printany is called directly when printing panics
+		// 2. An additional case statement that handles printing runtime.hex
+		//    values. Without this case statement, the default case will print
+		//    the runtime.hex values in a way not consistent with normal panic
+		//    outputs
+		for _, decl := range file.Decls {
+			funcDecl, ok := decl.(*ast.FuncDecl)
+			if ok && funcDecl.Name.Name == "printany" {
+				for _, stmt := range funcDecl.Body.List {
+					switchStmt, ok := stmt.(*ast.TypeSwitchStmt)
+					if !ok {
+						continue
+					}
+
+					switchStmt.Body.List = append(switchStmt.Body.List, printanyHexCase)
+					break
+				}
+
+				funcDecl.Body.List = append([]ast.Stmt{hidePanicsCheckStmt}, funcDecl.Body.List...)
+				break
+			}
+		}
+
+		file.Decls = append(file.Decls, panicprintDecl)
+	case "panic.go", "traceback.go":
+		// Change all calls to print, which we don't control, to
+		// panicprint, which we do control and does the same thing.
+		ast.Inspect(file, switchPanicPrints)
+	}
+}
+
+var hidePanicsCheckStmt = &ast.IfStmt{
+	Cond: &ast.Ident{
+		Name: "hidePanics",
+	},
```

Comment:
make all of these three-line simple Ident expressions a single line, which will make all of your AST code massively less verbose.